docs/docs/connectors/slack.mdx

```diff
@@ -164,6 +164,7 @@ slack:
   slack_retry_number_header: "x-slack-retry-num"  # Slack HTTP header name indicating the attempt number. This configuration is optional.
   errors_ignore_retry: None  # Any error codes given by Slack included in this list will be ignored. Error codes are listed [here](https://api.slack.com/events-api#errors).
   use_threads: False  # If set to True, bot responses will appear as a threaded message in Slack. This configuration is optional and set to False by default.
+  conversation_granularity: "sender" # sender allows 1 conversation per user (across channels), channel allows 1 conversation per user per channel, thread allows 1 conversation per user per thread. This configuration is optional and set to sender by default.
```

Comment:
@tayfun @srinivasupadhya  ,  Shall we add something like "In case a retrieved message does not specify the necessary information for the required level of granularity, then Rasa will fall back to use the next possible level of granularity and use the respective information as 'sender_id' (e.g. if `thread_id` is set to `None` and a `channel` is specified, while `conversation_granularity` is set to `thread`, then the `sender_id` will be a combination of the `user` and `channel` information only)."   -- It is kind of obvious that we can't get/use that info if it isn't there. But just to raise awareness that this might happen?

Reply:
Might be a good idea, makes it more clear for sure.

Reply:
This doesn't seem to happen for a slack message, but happened during a webhook validation on slack app setup page (Event Subscriptions). If channel / thread data is missing, responding to the message will not be possible & hence it is unlikely to happen. I think we can skip this.